signer/src/testing/transaction_coordinator.rs

```diff
@@ -19,87 +26,71 @@ use crate::transaction_coordinator;
 use rand::SeedableRng as _;
 use sha2::Digest as _;
 
-struct EventLoopHarness<S, C> {
-    event_loop: EventLoop<S, C>,
-    block_observer_notification_tx: tokio::sync::watch::Sender<()>,
-    storage: S,
+use super::context::TestContext;
+use super::context::WrappedMock;
+
+struct EventLoopHarness<C> {
+    event_loop: EventLoop<C>,
+    context: C,
+    is_started: Arc<AtomicBool>,
 }
 
-impl<S, C> EventLoopHarness<S, C>
+impl<C> EventLoopHarness<C>
 where
-    S: storage::DbRead + storage::DbWrite + Clone + Send + 'static,
-    C: crate::bitcoin::BitcoinInteract + Send + 'static,
+    C: Context + 'static,
 {
     fn create(
+        context: C,
         network: network::in_memory::MpmcBroadcaster,
-        storage: S,
-        bitcoin_client: C,
         context_window: usize,
         private_key: PrivateKey,
         threshold: u16,
     ) -> Self {
-        let (block_observer_notification_tx, block_observer_notifications) =
-            tokio::sync::watch::channel(());
-
         Self {
             event_loop: transaction_coordinator::TxCoordinatorEventLoop {
-                storage: storage.clone(),
+                context: context.clone(),
                 network,
-                block_observer_notifications,
                 private_key,
                 context_window,
                 threshold,
-                bitcoin_client,
                 bitcoin_network: bitcoin::Network::Testnet,
                 signing_round_max_duration: Duration::from_secs(10),
             },
-            block_observer_notification_tx,
-            storage,
+            context,
+            is_started: Arc::new(AtomicBool::new(false)),
         }
     }
 
-    pub fn start(self) -> RunningEventLoopHandle<S> {
-        let block_observer_notification_tx = self.block_observer_notification_tx;
-        let join_handle = tokio::spawn(async { self.event_loop.run().await });
-        let storage = self.storage;
+    pub async fn start(self) -> RunningEventLoopHandle<C> {
+        let is_started = self.is_started.clone();
+        let join_handle = tokio::spawn(async move {
+            is_started.store(true, Ordering::SeqCst);
+            self.event_loop.run().await
+        });
+
+        while !self.is_started.load(Ordering::SeqCst) {
+            tokio::time::sleep(Duration::from_millis(10)).await;
+        }
```

Comment:
Ahhh, this is to eliminate the race condition?

Reply:
Yeah -- could probably have used a Notify channel like I did elsewhere but this is what I ended up with here 🤷🏻‍♂️